lib/teleterm/services/desktop/directorysharing_test.go

```diff
@@ -0,0 +1,176 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package desktop
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOpenSharedDirectory(t *testing.T) {
+	path := t.TempDir()
+	filePath := filepath.Join(path, testFilename)
+	err := os.WriteFile(filePath, []byte("test"), 0600)
+	require.NoError(t, err)
+	access := DirectoryAccess{}
+	err = access.Open(filePath)
+	require.True(t, trace.IsBadParameter(err), "%q is not a directory", filePath)
+}
+
+const (
+	testDirname         = "test_dir"
+	testFilename        = "test_file"
+	testSymlinkFilename = "test_symlink"
+)
+
+func setUpSharedDir(t *testing.T) (*DirectoryAccess, string) {
+	path := t.TempDir()
```

Comment:
```suggestion
	t.Helper()
	path := t.TempDir()
```

This is a shared setup helper, so `t.Helper()` will make sure that the line number reported in test failures is the line that calls `setUpSharedDir`, not the failure inside this function.